modules/engine/src/shader-inputs.ts

```diff
@@ -24,8 +24,7 @@ export class ShaderInputs<
    * The map of modules
    * @todo should should this include the resolved dependencies?
    */
-  // @ts-expect-error Fix typings
-  modules: Readonly<{[P in keyof ShaderPropsT]: ShaderModule<ShaderPropsT[P]>}>;
+  modules: Readonly<{[P in keyof ShaderPropsT]: ShaderModule<Record<string, any>>}>;
```

Comment:
Making this change throws away the knowledge of the type the props of a `ShaderModule` has, which seems like a shame. 

Perhaps we are trying to be too clever with the types here though and for now we can go with `ShaderModule<Record<string, unknown>>`, what do you think @ibgreen? Realistically, I'm not going to have the bandwidth to look at this for a few weeks

Reply:
Agree that having types for the shader module props is a major improvement in v9 that we don't want to give up without a deep understanding of what is going wrong. 

What we could perhaps consider is landing this on the `9.0-release` branch instead of master. But still I would much rather we root cause and fix the issue rather than just remove typing.